docs/core-services/container-orchestration-image-auth.md

```diff
@@ -0,0 +1,136 @@
+
+## High-level flow
+
+The Kura container authenticity and allowlist enforcement feature is designed to address container authenticity concerns by providing a mechanism to perform the signature verification of container images and restricting which container images can be deployed on the Kura platform.
+
+This is achieved by implementing the following flow:
+
+![](./images/container-auth-high-level-flow.png){ width="600" }
+
+Description of the flow:
+
+- **Monitoring**: Kura monitors the container engine events. When a "container start" event is triggered it intercepts the information regarding the image used to spin up the container and starts performing the check.
+- **Allowlist** Firstly it checks if the digest of the image used to run the container can be found in the Kura allowlist. If so, the container is allowed to run. If not, Kura proceeds with the image signature verification.
+- **Signature verification**: If no trust anchor (i.e. a public key or a X509 certificate depending on the signature mechanism) is available for the verification process to happen, the container is not allowed to run. If one is available, the image signature is verified and the allowlist is updated accordingly.
+
+This flow applies to both containers managed by Kura itself and containers ran directly by the user via CLI and allows for a fine-grained control over the container images that can be deployed on the Kura platform.
+
+- **Unmanaged containers**: Kura allows the user to define a list of container images that can be deployed on the platform. This is done by adding the image digests to the allowlist. When a container is started, Kura checks if the image digest is in the allowlist and, if so, allows the container to run. This is meant to be used by users who, for any reason, need to run a container _outside_ the Kura framework but still want the safety guarantees provided by pinning the images to a specific digest.
+- **Managed containers**: Kura also allows the user, for containers ran by Kura itself, to provide a trust anchor to be used for the signature verification process. This allows the user to use a _mutable_ tag when specifying the container image, without giving up the required authenticity checks. Onche the image is verified its digest is stored within the Kura allowlist allowing it to be ran both from Kura and the CLI without the need for an internet connection. When a new image is published the user can simply trigger the pull and the verification process will update the digest automatically.
+
+Note that the user can still directly provide an image digest when running a Kura-managed container, bypassing the signature verification process. This is meant to be used when the user is sure about the image authenticity and does not want to go through the verification process or the image was not signed.
```

Comment:
I'm answering this one since I originally wrote this.

The idea here is to explain the fact that the user can set directly the image digest in case:
- they already **trust** the image and thus don't require the signing verification process to be performed (example scenario: they downloaded and performed the image signature verification on their workstation and thus already know it can be trusted)
- the image was not signed in the first place but they still **trust** that image

A better way to put this then would be:

> This feature is meant to be used when
> -  the user already ensured the image authenticity and does not want to go through the verification process again
> - the image was not signed but the user still wants the enforcement trust guarantees provided by Eclipse Kura

What do you think?

Reply:
Addressed in d7d9bc47d06b1768e30c397aeb278882ac4b0d9b